### PR TITLE
KIL-311: Add format validation for eval score names

### PR DESCRIPTION
**Pull request by sfierro**
## Summary
This PR fixes a bug where eval score names with invalid formatting (e.g., trailing whitespace like "Correctness ") would cause validation errors when used to create TaskRequirement objects in the rating UI.

## Changes
- Changed `EvalOutputScore.name` from `str` to `FilenameStringShort`
- This enforces the same validation as `TaskRequirement.name`:
  - Max 32 characters
  - No forbidden characters (/ \ ? % * : | < > . , ; = newline)
  - No leading/trailing whitespace or underscores
  - No consecutive whitespace or underscores
- Added comprehensive tests for the validation

## Testing
- All existing tests pass
- Added new test `test_eval_output_score_name_validation` that specifically tests the validation rules
- Verified the exact error case from the ticket ("Correctness " with trailing space) is now caught

Fixes Linear ticket KIL-311

## Summary by CodeRabbit

* **Bug Fixes**
  * Stricter validation for evaluation score and evaluator names to block invalid characters, excessive length, and disallowed spacing/underscore patterns.
  * Name inputs are now trimmed and validation errors short-circuit earlier; submission state and user-facing error messages reflect validation failures.

* **New Features**
  * New client-side filename-style validator for evaluator creation with clearer, name-specific error messages and consistent rules.

* **Tests**
  * Added tests covering name validation rules and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: app/web_ui/src/routes/(app)/evals/[project_id]/[task_id]/create_evaluator/select_eval_template.svelte [hunk 1]
```diff
@@ -444,6 +445,7 @@
         return
       }
 
+      tool_call_eval_name = tool_call_eval_name.trim()
```
Comment (chiang-daniel) [inline app/web_ui/src/routes/(app)/evals/[project_id]/[task_id]/create_evaluator/select_eval_template.svelte]: Should we run this in the validator code in the backend and return error? I think there are other formats that were not allowed according to the error log. 
```
Name is invalid. The name cannot contain any of the following characters: / \\ ? % * : | < > . , ; = \\n, consecutive whitespace/underscores, or leading/trailing whitespace/underscores
```

Comment (sfierro) [reply app/web_ui/src/routes/(app)/evals/[project_id]/[task_id]/create_evaluator/select_eval_template.svelte]: updated to use a proper validator with a comment to stay in sync with backend
